fetch.bs

```diff
@@ -2204,6 +2204,39 @@ unset or <a for=request>keepalive</a> is false, <a lt=terminated for=fetch>termi
 <a for="fetch group">fetch record</a>'s <a for="fetch record">fetch</a>.
 
 
+<h3 id=resolving-domains>Resolving domains</h3>
+
+<p>To <dfn>resolve a domain</dfn>, given a <a for=/>domain</a> <var>domain</var> and a
+<a for=/>network partition key</a> <var>key</var>, perform an <a>implementation-defined</a>
+operation to turn <var>domain</var> into a <a for=/>set</a> of one or more
+<a for=/>IP addresses</a>. If this operation succeeds, return the <a for=/>set</a> of
+<a for=/>IP addresses</a>. If it fails, return failure. The results of this operation may be cached.
+If they are cached, <var>key</var> must be used as part of the cache key.
+
+<p class=note>Typically this operation would involve DNS. The particulars (apart from the cache key)
+are not tied down as they are not pertinent to the system the Fetch Standard establishes. Other
+documents ought not to build on this primitive without having a considered discussion with the Fetch
+Standard community first. [[RFC1035]]
+
+<p>To <dfn>resolve an origin</dfn>, given an <a for=/>origin</a> <var>origin</var> and a
+<a for=/>network partition key</a> <var>key</var>:
+<!-- Should we assert the scheme here to be an HTTP(S) scheme or a WebRTC scheme? -->
+
+<ol>
+ <li><p>If <var>origin</var>'s <a for=origin>host</a> is an <a for=/>IP address</a>, then return
+ « <var>origin</var>'s <a for=origin>host</a> ».
```

Comment:
Note that IP Address is defined as a 32-bit or 128-bit unsigned integer.  Neither of those are a valid host.  Is there some host to IP address mapping we should reference in the line above and here?

Reply:
In spec-theory an origin's host that is an IP address is a 32-bit or 128-bit integer. As we don't define the actual connection-creation process, how that integer is serialized isn't referenced either (though we could reference it I suppose, https://url.spec.whatwg.org/#concept-ipv4-serializer and https://url.spec.whatwg.org/#concept-ipv6-serializer defines these.

Reply:
Maybe just add "as an IP address", then?  Or perhaps I'm alone in thinking pseudo-code as text should be type safe.  :)

Reply:
What part is not type safe here though? An origin's host is a host, which is roughly speaking a domain or an IP address. Asking if it's one or the other should be okay?

Reply:
Well, we're returning, e.g., "127.0.0.1" or "[::1]".  Neither of those fit the linked definition of "IP address" of an unsigned integer.

Reply:
But the spec doesn't, right? That you implement them as a different type is okay, as long as the results are indistinguishable.

Reply:
I'm talking about this spec, not Chrome's code.  It says return the host.

Reply:
Oh, host is defined as a domain or an IP address, as opposed to as the raw representation from the URL...ok, the magic is behind extracting the host from the URL, not the IP address from the host.

Reply:
Defined parser magic, yes :-)